### ISSUE TITLE
New Rule: no-reversed-links

### ISSUE THREAD

**Issue by xbinaryx** [eslint/markdown#386]
### Rule details

This rule is triggered when text that appears to be a link is encountered, but where the syntax appears to have been reversed (the [] and () are reversed)

### What type of rule is this?

Warns about a potential problem

### Example code

```markdown
(Incorrect link syntax)[https://www.example.com/]
```

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

Reversed links are not rendered as usable links.

**Comment by lumirlumir** [comment]
I also think that gathering more feedback about this rule would be helpful.
